salt/modules/keystore.py

```diff
@@ -206,3 +207,25 @@ def remove(name, keystore, passphrase):
     else:
         # No alias found, notify user
         return False
+
+
+def get_sha1(certificate):
+    """
+    Returns the SHA1 sum of a ASN1/PEM certificate
+
+    :param name: ASN1/PEM certificate
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' keystore.get_sha1 "(certificate_content_string)"
+
+    """
+    ASN1 = OpenSSL.crypto.FILETYPE_ASN1
+    PEM = OpenSSL.crypto.FILETYPE_PEM
+    if certificate[0] == 48:
```

Comment:
**suggestion** seeing this in a second place it woul make sense to just have a detect encoding function...

Reply:
It could easily return PEM or ASN1